src/app/screens/ledger/importLedgerAccount/index.tsx

```diff
@@ -276,22 +276,43 @@ function ImportLedger(): JSX.Element {
     },
   });
 
+  const getNewAccountId = () => {
+    if (ledgerAccountsList.length === 0) {
+      return 0;
+    }
+
+    return ledgerAccountsList[ledgerAccountsList.length - 1].id + 1;
+  };
+
+  const getDeviceNewAccountIndex = (masterKey?: string) => {
+    const ledgerAccountsIndexList = ledgerAccountsList
+      .filter((account) => masterKey === account.masterPubKey)
+      .map((account) => account.deviceAccountIndex);
+
+    for (let i = 0; i < ledgerAccountsIndexList.length; i += 1) {
+      if (ledgerAccountsIndexList[i] !== i) {
+        return i;
+      }
+    }
+
+    // If no missing number is found, return the length of the array
+    return ledgerAccountsIndexList.length;
+  };
+
```

Comment:
these two functions are good candidates for being converted to pure functions and separated from the 'screen' component.

this file is approaching 1000 lines long, and will be very difficult for any new dev to make changes. I'd suggest we separate concerns here between presentation and implementation logic, e.g. you could use another file to contain ledger helper functions, or create a custom hook to abstract ledger implementation details. also think about making unit testing easier

Reply:
I agree, added this suggestion to the refactoring task https://linear.app/xverseapp/issue/ENG-2537/refactor-ledger-related-logic if you don't mind